### PR TITLE
Nanite removal superbranch -> dev

### PR DESCRIPTION
**Pull request by Turtlelord26**
Closing out the Nanite Removal project, wherein the nanite aesthetic was reimagined into a more soft-sci-fi feel than its former definitely-space-magic flavor.

Former uses of the nanite bar have been rewritten using a new resource: Energy. Energy is stored by equipment that uses it and also batteries. Generally cloaking, shields, and other armor hard point modules.
As part of this, the Engineer resource Energy has been renamed to Potential.

We introduce Medical Nanites, the new nanites. Medical nanites are now used to power Active Feats based on a multi-cooldown system. Each Active Feat puts a unit of medical nanites on cooldown for a duration, but characters may have multiple units at their disposal.
(Dev note: we can make higher-level feat content very splashy by using multiple units instead of one)

Resting has gained rules text, and applies healing based on medical nanites.

Concurrent to this superbranch, class feats have been reworked to remove old nanites (thanks @bleehu), and some older feats that used nanites have been deleted. The former is already in dev, the latter will merge following this PR.

I'll make a full technical review with the nitty gritty.

### CODE REVIEW

File: CharacterCreation/Class-Specific Documentation/Doctor Procedures.txt [hunk 1]
```diff
@@ -802,12 +802,12 @@ Effect:			Target gets a 30 point capacitive shield for the duration of
...
+Adrenaline Reward:	10 Points
 Range:			Touch
-Action:			Free Action
-Effect:			Transfer n number of Adrenaline points from the Doc to the 
-patient. If This trick is used on another doctor, no adrenaline points are 
-earned. 
+Action:			1 Action
+Effect:			Choose 1. Remove up to 3 Rounds of cooldown from one unit of 
+Medical Nanites owned by the target, or the cooldown of the target's next active 
+feat before your next turn is reduced by up to 3 Rounds, to a minimum of zero.
```
Comment (bleehu) [inline CharacterCreation/Class-Specific Documentation/Doctor Procedures.txt:810]: looking forward to testing this!

File: Items/05_01_Hard_Point_Equipment.txt [hunk 2]
```diff
@@ -0,0 +1,242 @@
+==============================
+Hard Point Equipment
+==============================
```
Comment (bleehu) [inline Items/05_01_Hard_Point_Equipment.txt:3]: We'll have to add a new document to the website to display this.

**Review by bleehu — APPROVED** [review]
Shiptit! Way to go, :turtle: !!!

**Review by Turtlelord26 — COMMENTED** [review]
Changes by File

## Basic Rules

-Top-level descriptions of classes, nanites, and energy (new) altered for the new aesthetic. Nonfunctional. ✔️ 
-The maximum nanites by level section has been terminated. Nonfunctional. ✔️ 
-Added resting mechanics. Characters now (officially) heal during rests, based on medical naniteness and level. Rests are rate-limited as well. Functional. Testable. ✔️ 
-Feat description prose changes. Descriptive text, nonfunctional. ✔️ 
-Active feat rules now require the use of medical nanites, using a cooldown system. functional. testable. ✔️

## Leveling Up

-Added a column describing maximum number of units of medical nanites per level. Functional. Numbers testable. ✔️

## Cloaking Rules

-Cloaking now uses a new resource "Energy" instead of nanites. Activation and upkeep costs remain, just with the new resource name. Technically changing the name of a resource is not a functional change, but the resource economy was altered, see below.
-Removed free cloaking for taking no actions as out of aesthetic. Now instead cloaking is reduced cost while not taking actions. Functional, testable. ✔️
-Added an Overheat stat to cloak modules, which puts a cap on how long a cloak can be held. Replaces the old cloak turn limit by level (which has been removed from the cloaking table) - ie the limit is tied to equipment instead of level now. Aesthetic. Functional, testable. ✔️
-Cloaking modules now have a built-in resource supply, compensating for the loss of the baseline resources from every character's nanite bar. functional, testable ✔️
-Removed the one-cloak-per-encounter limit for early levels. Functional, testable. ✔️

## Classes

-removed primary attributes. With the destruction of the old nanites, these were vestigial, so deleting them is actually not a functional change. ✔️
-The former Engineer resource Energy has been renamed to Potential, specifically to free up the name for the above-mentioned resource. Nonfunctional. ✔️

## Feats

-Active feat economy has been switched off old-nanites resource cost and onto new-nanites cooldowns. They have ben given durations (how long it takes to execute the feat) and cooldowns (how long your unit(s) of nanites need to recharge afterward). Functional, testable. Testing indicated for every feat. ✔️
-Some functional changes resulted from the new system:
--Adrenaline rush can only move 4 or 8 extra meters, instead of 1-10. Cannot be taken twice anymore.
--Aimed shot is now considered a crit instead of automatically critting, but has bonus damage to compensate.
--Bunker, Multi-target are deleted.
--Cover slide gains an evasion buff vs readied attacks.
--Duck, Bob, and Weave has been unintentionally deleted and will be reintroduced in a subsequent patch. 🎏🎏🎏🎏🎏🎏
--Flash Dance has been substantially reworked, and may be remotely balanced now.
--Gun Kata same.
--Kata now additionally triggers off certain combat maneuvers.
--Run 'n' gun's accuracy penalty is now cumulative per attack.

## Doctor Procedures

-Nanite Restore has been reworked to use the new nanites. Functional, testable. Testing indicated. ✔️ 🎏

## Engineer Procedures

-Further replacement of Energy by Potential, including flavor text and process costs. nonfunctional ✔️
-Enhanced flavor text for why munitions don't persist between encounters. nonfunctional. ✔️

## Hard Point Equipment

-Separated from Armor doc into its own file for ease of reference. Nonfunctional. ✔️
-Significant rebalancing of costs to account for the new Energy economy. Functional, basically every item requires testing. ✔️
-New items: battery packs. Allow you to add more Energy storage at the cost of hard points - allowing characters to choose between doing one thing strongly or having more utility. Functional, testable. Testing indicated. ✔️ 🎏

## Armors

-Hard point equipment removed, nonfunctional. ✔️

## Items

-Added Medical Nanite Items. They are functional and testable ✔️
--Medical Nanites Booster Shot: A partial version, supplies less rest healing and no active feats.
--Medical Nanites Unit: The full version. maximal rest healing and they let you flash dance.
-Reworked Nanite Pack to use medical nanites. Very similar to nanite restore. functional, testable. testing indicated. ✔️ 🎏

**Comment by bleehu** [comment]
Um, why are we removing the feats section from basic rules?

**Comment by Turtlelord26** [comment]
It's a duplicate. 
https://github.com/bleehu/Compound_X/pull/804/files#diff-612d190f94ea71bed2a608f9b26c9cbcR548
This is still there.

**Comment by bleehu** [comment]
Carry on!

**Review by rosebrallier — APPROVED** [review]
Shipit!!!

**Pull request merged**